protocol/rpcprovider/rpcprovider.go

```diff
@@ -264,6 +264,15 @@ func (rpcp *RPCProvider) SetupProviderEndpoints(rpcProviderEndpoints []*lavasess
 	return disabledEndpointsList
 }
 
+func (rpcp *RPCProvider) getAllAddonsAndExtensionsFromNodeUrlSlice(nodeUrls []common.NodeUrl) []string {
+	allNodeUrlAddonsAndExtensions := []string{}
+	for _, nodeUrl := range nodeUrls {
+		allNodeUrlAddonsAndExtensions = append(allNodeUrlAddonsAndExtensions, nodeUrl.Addons...)
+	}
+	utils.LavaFormatDebug("Adding supported addons and extensions to chain parser", utils.LogAttr("info", allNodeUrlAddonsAndExtensions))
+	return allNodeUrlAddonsAndExtensions
```

Comment:
return a policy object here that is prepared from the addons and extensions the provider configured